librz/analysis/esil.c

```diff
@@ -560,12 +560,14 @@ RZ_API int rz_analysis_esil_signext(RzAnalysisEsil *esil, bool assign) {
 
 	char *p_dst = rz_analysis_esil_pop(esil);
 	if (!p_dst) {
+		free(p_src);
```

Comment:
may I suggest that a `goto` here would fit very well to avoid forgetting stuff as they are added?